src/src/WebServer/JSON.cpp

```diff
@@ -524,9 +524,9 @@ void handle_buildinfo() {
 void stream_to_json_value(const String& value) {
   NumericalType detectedType;
   bool isNum  = isNumerical(value, detectedType);
-  bool isBool = (Settings.JSONBoolWithQuotes() && ((value.equalsIgnoreCase(F("true")) || value.equalsIgnoreCase(F("false")))));
+  bool isBool = (Settings.JSONBoolWithoutQuotes() && ((value.equalsIgnoreCase(F("true")) || value.equalsIgnoreCase(F("false")))));
 
-  if (((value.length() == 0) || !isNum || mustConsiderAsString(detectedType)) && !isBool) {
+  if (!isBool && ((value.length() == 0) || !isNum || mustConsiderAsString(detectedType))) {
```

Comment:
This code is tested to work?
Looks OK, which means the last version could not have been working.

Reply:
I've moved the `!isBool` part from the end of the condition to the start, since the last commit, to avoid calling `mustConsiderAsString()` every time, as that seems quite expensive.
The original condition looked OK when first found, but I haven't been able to really test that part (yet).